src/models/model.cpp

```diff
@@ -2632,11 +2632,57 @@ bool model::load_from_checkpoint_distributed(persist& p)
   return true;
 }
 
-void model::write_proto(lbann_data::Model* proto)
+void model::write_proto(lbann_data::Model& proto)
 {
-  proto->Clear();
-  // if (m_comm->am_world_master())
-  //   proto->set_mini_batch_size(m_max_mini_batch_size);
+  if (!get_comm()->am_trainer_master())
+    return;
+
+  proto.Clear();
+  proto.set_name(this->get_name());
+  this->get_objective_function()->write_proto(*proto.mutable_objective_function());
+  for (auto const* metric : this->get_metrics()){
+    auto* met = proto.add_metric()->mutable_layer_metric();
+    met->set_name(metric->name());
+    met->set_unit(metric->get_unit());
+  }
+  //FIXME(KLG): Data layout is a single string but I have to get it from the layers?
+  auto layout = this->get_layer(0).get_data_layout();
+  switch (layout) {
+  case data_layout::DATA_PARALLEL:
+    proto.set_data_layout("data_parallel");
+    break;
+  case data_layout::MODEL_PARALLEL:
+    proto.set_data_layout("model_parallel");
+    break;
+  default:
+    LBANN_ERROR("Invalid data layout");
+  }
```

Comment:
Remove from here. This allows a user to set a default data layout for their model, which can be a useful short-hand. However, by the time this function can be called, each layer in the model will have established its data layout, so a default is not needed anymore.